tools/src/test/java/org/apache/kafka/tools/consumer/group/ConsumerGroupExecutor.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.tools.consumer.group;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+public class ConsumerGroupExecutor implements AutoCloseable {
+    final int numThreads;
+    final ExecutorService executor;
+    final List<ConsumerRunnable> consumers = new ArrayList<>();
+
+    public ConsumerGroupExecutor(
+            String broker,
+            int numConsumers,
+            String groupId,
+            String groupProtocol,
+            String topic,
+            String strategy,
+            Optional<String> remoteAssignor,
+            Optional<Properties> customPropsOpt,
+            boolean syncCommit
```

Comment:
We have 2 well defined group types, classic and consumer, with different creation params, and having them all together seems a bit confusing (ex. having assignment strategy along with remote assignor)...not very intuitive when creating a new instance (that I expect several other test files will do as they get migrated). So wonder if it would maybe be clearer to split them into static builders that would take only the params they need? like:
```
public static ConsumerGroupExecutor forConsumerGroup(
                                                         String broker,
                                                         int numConsumers,
                                                         String groupId,
                                                         String topic,
                                                         Optional<String> remoteAssignor,
                                                         Optional<Properties> customPropsOpt,
                                                         boolean syncCommit) {

    }

    public static ConsumerGroupExecutor forClassicGroup(
                                                        String broker,
                                                        int numConsumers,
                                                        String groupId,
                                                        String topic,
                                                        String strategy,
                                                        Optional<Properties> customPropsOpt,
                                                        boolean syncCommit) {

    }
```

Reply:
Indeed, too many parameters can be very confusing for users. I will make the changes according to your suggestion. Thank you! 😀